### PR TITLE
Level Setup

### LINKED ISSUE #83: Room Gameplay
TODO: Update specific definition.

### LINKED ISSUE #85: Consistent behaviors across rooms
We need to establish room transitions and ensure that values are kept correctly across these transitions.

Blocked by #15.

### LINKED ISSUE #111: Set up different room types and scripts used in it

### LINKED ISSUE #118: Inventory changes stay the same when moving through scene

### PR DESCRIPTION
**Pull request by Galaxy25**
Closes #83 
Closes #85 
Closes #111 
Closes #118

### CODE REVIEW

File: Assets/Scripts/Player/WeaponInventory.cs [hunk 1]
```diff
@@ -12,14 +12,10 @@ public class WeaponInventory : MonoBehaviour
...
-    public ListOfWeapons Weapons;
-
-    [System.Serializable, HideInInspector]
-    public class ListOfWeapons : ReorderableArray<WeaponData>
-    {
-    }
+    public List<WeaponData> NewWeapons;
+    private static List<WeaponData> Weapons = new List<WeaponData>();
 
+    private bool hasAddedWeapons = false;
```
Comment (BrandonPacewic) [inline Assets/Scripts/Player/WeaponInventory.cs:18]: Can you explain what this is used for?

Comment (Galaxy25) [reply Assets/Scripts/Player/WeaponInventory.cs:18]: Its to avoid `Start()` repeatable adding weapons each time the player moves through   scene

Comment (BrandonPacewic) [reply Assets/Scripts/Player/WeaponInventory.cs:18]: Would this be fixed if we used `Awake()` instead of `Start()`? Or is this the best option?

Comment (Galaxy25) [reply Assets/Scripts/Player/WeaponInventory.cs:18]: No it wouldn't change the scenes

Comment (BrandonPacewic) [reply Assets/Scripts/Player/WeaponInventory.cs:18]: 👍

**Comment by BrandonPacewic** [comment]
@Galaxy25 I pushed a change that makes it so that the map will be wiped each time you press the start button on the main menu. Please verify that I didn't break anything. 🙃

**Comment by Galaxy25** [comment]
@BrandonPacewic Everything works

**Review by BrandonPacewic — APPROVED** [review]
I'm getting ready to merge and updating the change log - please @me if any other changes are pushed.

**Pull request merged**